Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1170 @@
+        ):
+            print(
+                f"{t.yellow}Looking on pr number [{pr.number}]: last updated: "
+                f"{str(pr.updated_at)}, branch={pr.head.ref}"
+            )
+
+            conditions = [
+                LastModifiedCondition(pr=pr, git_repo=self.git_repo_obj),
+                LabelCondition(pr=pr, git_repo=self.git_repo_obj),
+                AddedRNFilesCondition(pr=pr, git_repo=self.git_repo_obj),
+                HasConflictOnAllowedFilesCondition(pr=pr, git_repo=self.git_repo_obj),
+            ]
+            for c1, c2 in pairwise(conditions):
+                c1.set_next_condition(c2)
+
+            base_cond_result = conditions[0].check()
+            if base_cond_result.should_skip:
+                continue
+
+            packs_to_autobump = []
+            conflicting_packs = base_cond_result.conflicting_packs or set()
+            for pack in conflicting_packs:
+                origin_md, branch_md, pr_base_md = MetadataCondition.get_metadata_files(
+                    pack_id=pack,
+                    pr=pr,
+                    git_repo=self.git_repo_obj,
+                )
+                conditions = [
+                    PackSupportCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                    ),
+                    MajorChangeCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                        origin_base_metadata=origin_md,
+                    ),
+                    MaxVersionCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                        origin_base_metadata=origin_md,
+                    ),
+                    OnlyVersionChangedCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                        origin_base_metadata=origin_md,
+                    ),
+                    OnlyOneRNPerPackCondition(
+                        pack=pack, pr=pr, git_repo=self.git_repo_obj
+                    ),
+                    SameRNMetadataVersionCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                    ),
+                    AllowedBumpCondition(
+                        pack=pack,
+                        pr=pr,
+                        git_repo=self.git_repo_obj,
+                        branch_metadata=branch_md,
+                        pr_base_metadata=pr_base_md,
```

Comment:
this part is worth its own method

Reply:
I am not opening the metadata file 7 times. I am opening it one time and save it as a dict and give to every condition the dictionary.

Reply:
I added a little change to this function.